mnemosyne/batch_tool.py

```
@@ -0,0 +1,199 @@
+
+def batch_validation_error_payload(exc: BatchValidationError) -> dict[str, Any]:
+    payload: dict[str, Any] = {"status": "error", "error": str(exc)}
+    if exc.failed_index is not None:
+        payload["failed_index"] = exc.failed_index
+    if exc.action:
+        payload["action"] = exc.action
+    return payload
+
+
+def apply_beam_batch(
+    beam: Any,
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `beam`

(ANN401)